crates/rsonpath-lib/src/engine/main.rs

```diff
@@ -240,13 +250,32 @@ impl<'q, 'b, I: Input> Executor<'q, 'b, I> {
         R: QueryResult,
     {
         self.next_event = classifier.next();
+        if self.is_list {
+            if self.array_count.try_increment().is_err() {
+                // We can't possibly match these items, but it's not necessarily an error either.
+                return Ok(());
+            }
```

Comment:
While we can't match those with an _index_ selector, there's nothing in either standard (JSONPath or I-JSON) that would prohibit us from matching them with the _wildcard_ selector.

This is a very niche case and quite hard to write a test for, let's try to be 100% correct here.

Reply:
You're absolutely correct.  I was mistaken that the wrapping `is_list` would catch those, but it definitely doesn't.

Reply:
I've moved this check to after the wildcard match.